File: intervals/56_merge_intervals.py
def merge(intervals: list[list[int]]) -> list[list[int]]:
    intervals.sort()

    result = [intervals[0]]

    for i in range(1, len(intervals)):
        a1, b1 = result[-1]
        a2, b2 = intervals[i]

        if max(a1, a2) <= min(b1, b2):
            result[-1] = [min(a1, a2), max(b1, b2)]
        else:
            result.append([a2, b2])

    return result

File: intervals/56_merge_intervals.go
func merge(intervals [][]int) [][]int {
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	result := [][]int{intervals[0]}

	for i := 1; i < len(intervals); i++ {
		prev := result[len(result)-1]
		start := intervals[i][0]
		end := intervals[i][1]

		if prev[1] >= start {
			prev[1] = max(prev[1], end)
		} else {
			result = append(result, intervals[i])
		}
	}

	return result
}